src/porepy/utils/ui_and_logging.py

```diff
@@ -0,0 +1,48 @@
+"""This module contains functionality to make the progress bars work correctly with
+logging."""
+from __future__ import annotations
+
+import logging
+from contextlib import contextmanager
+from typing import Iterator, Optional, Type
+
+# Avoid some mpy trouble.
+from tqdm.autonotebook import tqdm as std_tqdm  # type: ignore
+from tqdm.contrib.logging import _TqdmLoggingHandler  # type: ignore
+from tqdm.contrib.logging import logging_redirect_tqdm  # type: ignore
+from tqdm.contrib.logging import (  # type: ignore
+    _get_first_found_console_logging_handler,
+)
+
+
+@contextmanager
+def logging_redirect_tqdm_with_level(
+    loggers: Optional[list[logging.Logger]] = None,
+    tqdm_class: Type[std_tqdm] = std_tqdm,
+) -> Iterator[None]:
+    """Extend capability of ``tqdm.contrib.logging_redirect_tqdm`` s.t. the logging
+    handler level gets passed."""
```

Comment:
Please expand the explanation, including specifying the issue with the current status of `tqdm`.